internal/pkg/manifest/validate_env.go

```diff
@@ -81,6 +84,56 @@ func (cfg environmentVPCConfig) validate() error {
 	return nil
 }
 
+// validate returns nil if securityGroupRule has all the required parameters set.
+func (cfg SecurityGroupRule) validate() error {
+	if cfg.CidrIP == "" {
+		return fmt.Errorf(`cidr`)
+	}
+	if cfg.IpProtocol == "" {
+		return fmt.Errorf(`ip_protocol`)
+	}
+	return cfg.PortsConfig.validate()
+}
+
+// validate if ports are set.
+func (cfg PortsConfig) validate() error {
+	if cfg.IsEmpty() {
+		return fmt.Errorf(`ports`)
+	}
+	if !cfg.Ports.IsEmpty() {
+		return cfg.Ports.validate()
+	}
+	return nil
+}
+
+// validate checks if PortsRangeBand is set correctly.
+func (str PortsRangeBand) validate() error {
+	ports := strings.Split(string(str), "-")
+	if len(ports) != 2 {
+		return fmt.Errorf("invalid ports value %s. Should be in format of ${from_port}-${to_port}", string(str))
+	}
+	return nil
+}
+
+// validate returns nil if securityGroupConfig is configured correctly.
+func (cfg securityGroupConfig) validate() error {
+	for idx, ingress := range cfg.Ingress {
+		if err := ingress.validate(); err != nil {
+			return fmt.Errorf(`validate ingress[%d]: %w`, idx, &errFieldMustBeSpecified{
+				missingField: err.Error(),
+			})
```

Comment:
After addressing the feedback above of moving `errFieldMustBeSpecified` to the individual validate() methods, we can then do:
```suggestion
			return fmt.Errorf(`validate ingress[%d]: %w`, idx, err)
```